### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by IvanHahan**
Simplify preprocess_all and train scripts execution

### CODE REVIEW

File: scripts/count_hours.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def read_audio_and_text(inputs):
+    audio_path = inputs['file_path']
+    text = inputs['text']
+    language = inputs['language']
+    text = ' '.join(text.split())
+    for p in ',.:;?!-_':
+        text = text.replace(p, '')
+    text = text.lower().split()
```
Comment (akademi4eg) [inline scripts/count_hours.py]: No need to process text in any way.

File: scripts/count_hours.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    filename, language, text = line.split(args.delimiter)
+    inputs = {
+        'file_path': filename,
+        'text': text.strip(),
+        'language': language
+    }
+    try:
+        return read_audio_and_text(inputs)
+    except Exception as err:
+        print(str(err))
```
Comment (akademi4eg) [inline scripts/count_hours.py:26]: Print error and return is exactly the same as not catching the error. I'd just remove try-except block.

File: scripts/count_hours.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        return 0
+
+
+def main(args):
+    # window = int(SAMPLE_RATE * args.window / 1000.0)
```
Comment (akademi4eg) [inline scripts/count_hours.py]: Please remove commented code blocks. :)

File: scripts/count_hours.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+
+if __name__ == '__main__':
+    warnings.simplefilter(action='ignore', category=FutureWarning)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input_file', help='File with audio paths and texts.', required=True)
+    parser.add_argument('--step', help='Analysis window step in ms.', type=int, default=10)
+    parser.add_argument('--start', help='Index of example to start from', type=int, default=0)
+    parser.add_argument('--count', help='Maximal phrases count, -1 for all phrases', type=int, default=-1)
```
Comment (akademi4eg) [inline scripts/count_hours.py]: Maybe let's remove 'count' param? I think noone uses it (same for 'start' param).
@EugKar what do you think?

File: scripts/count_hours.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        return read_audio_and_text(inputs)
+    except Exception as err:
+        print(str(err))
+        return 0
+
+
+def main(args):
+    # window = int(SAMPLE_RATE * args.window / 1000.0)
+    # step = int(SAMPLE_RATE * args.step / 1000.0)
+    lines = open(args.input_file, 'r').readlines()
```
Comment (akademi4eg) [inline scripts/count_hours.py]: This could be an issue for huge datasets, let's just use `for x in open(...):`.

**Review by akademi4eg — CHANGES_REQUESTED** [review]

File: scripts/count_hours.py (hunk 1 above)
Comment (IvanHahan) [reply scripts/count_hours.py]: snippet copied from preprocess_all). Should I remove it there as well?

File: scripts/count_hours.py (hunk 2 above)
Comment (IvanHahan) [reply scripts/count_hours.py:26]: It's done to prevent crash. This try-catch block would just skip faulty inputs

**Review by akademi4eg — APPROVED** [review]

**Pull request merged**